app/gui/src/project-view/components/visualizations/ScatterplotVisualization.vue

```diff
@@ -865,42 +867,49 @@ const makeSeriesLabelOptions = () => {
   return seriesOptions
 }
 
-config.setToolbar([
-  {
-    icon: 'select',
-    title: 'Enable Selection',
-    toggle: selectionEnabled,
-  },
-  {
-    icon: 'show_all',
-    title: 'Fit All',
-    onClick: () => zoomToSelected(false),
-  },
-  {
-    icon: 'zoom',
-    title: 'Zoom to Selected',
-    disabled: () => brushExtent.value == null,
-    onClick: zoomToSelected,
-  },
-  {
-    icon: 'add_to_graph_editor',
-    title: 'Create component of selected points',
-    disabled: () => !createNewFilterNodeEnabled.value,
-    onClick: createNewFilterNode,
-  },
-  {
-    type: 'textSelectionMenu',
-    selectedTextOption: yAxisSelected,
-    title: 'Choose Y Axis Label',
-    heading: 'Y Axis Label: ',
-    options: {
-      none: {
-        label: 'No Label',
-      },
-      ...makeSeriesLabelOptions(),
+const createTextSelectionButton = (): ToolbarItem => ({
+  type: 'textSelectionMenu',
+  selectedTextOption: yAxisSelected,
+  title: 'Choose Y Axis Label',
+  heading: 'Y Axis Label: ',
+  options: {
+    none: {
+      label: 'No Label',
     },
+    ...makeSeriesLabelOptions(),
   },
-])
+})
+
+function useScatterplotVizToolbar() {
+  const textSelectionButton = createTextSelectionButton()
+  return computed(() => [
+    {
+      icon: 'select' as Icon,
```

Comment:
`as` is not necessary here--there are typesafe ways to prevent the typechecker from widening these `icon` properties to `string`s. One approach would be to declare the return type, e.g. `computed<ToolbarItem[]>(...)`